docs/devcontainers.md

```diff
@@ -0,0 +1,164 @@
+# Devcontainer support
+
+A [development container](https://containers.dev) (or devcontainer for short) allows you to use a container as a development environment. It can be used to develop and run applications in the isolated environment that is persistent across different developers. This page describes the configuration for VS Code. You can refer to [this](https://containers.dev/supporting) page for other supported tools.
+
+## Requirements
+
+- Install [VS Code](https://code.visualstudio.com/docs/setup/setup-overview)
+- Install [Devcontainers Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+- Navigate to the root of Artery repository
+
+## Develop inside the container
+
+Working with devcontainers is very similar to [remote development](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) using __ssh__.  After installing devcontainer extension, you should observe
+an icon in the status bar of your editor that looks like two arrowheads pointing at each other. Clicking on that
+icon reveals a menu for remote connections, which should include an option for opening current directory
+inside a container:
+
+`Reopen in Container`
+
+Selecting this entry brings you a list of options for containers. The ones Artery provides are called _arch-linux_ and _debian_.
+You can choose any of them, but keep in mind that graphic applications (including Artery) might fail due to version difference
+between libraries installed in the container and in your host system.
+
+Same can be achieved with _Command palette_:
+
+- Press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>
+- Select `Dev Container: Reopen in Container`
+
+After choosing that, a VS Code window should reappear with connection established. If it's your first
+time opening, container will require to be built first, you can click on `view logs` in notification on the right
+side to track progress.
+
+Directory which you have initially opened will be mounted in the container and changes that you make to the source code 
+or other files in that directory are saved in your host's filesystem. Although keep in mind that build that you 
+created inside the container does not have to be compatible with you host system, so if you leave devcontainer you should
+run clear build.
```

Comment:
"if you leave devcontainer" suggests that clearing the build directory is required every time one closes the VSCode sessions.
My suggestion: "… inside the container might not be compatible with your host system. Please use distinct build directories for host and devcontainer builds if you want to build Artery in both environments."